### PR TITLE
Log expected & actual color in image access tests

### PR DESCRIPTION
**Pull request by Yay295**

### CODE REVIEW

File: Tests/test_image_access.py [hunk 1]
```diff
@@ -139,15 +139,18 @@ def check(self, mode, c=None):
...
         # check putpixel negative index
         im.putpixel((-1, -1), c)
-        assert (
-            im.getpixel((-1, -1)) == c
-        ), f"put/getpixel roundtrip negative index failed for mode {mode}, color {c}"
+        d = im.getpixel((-1, -1))
+        assert d == c, (
+            f"put/getpixel roundtrip negative index failed for mode {mode}, "
+            f"expected {c} got {d}"
+        )
```
Comment (hugovk) [inline Tests/test_image_access.py:153]:
```suggestion
        actual = im.getpixel((-1, -1))
        assert actual == c, (
            f"put/getpixel roundtrip negative index failed for mode {mode}, "
            f"expected {c} got {actual}"
        )
```

File: Tests/test_image_access.py [hunk 2]
```diff
@@ -166,13 +169,15 @@ def check(self, mode, c=None):
 
         # check initial color
         im = Image.new(mode, (1, 1), c)
-        assert (
-            im.getpixel((0, 0)) == c
-        ), f"initial color failed for mode {mode}, color {c} "
+        d = im.getpixel((0, 0))
+        assert d == c, f"initial color failed for mode {mode}, expected {c} got {d}"
```
Comment (hugovk) [inline Tests/test_image_access.py:173]:
```suggestion
        actual = im.getpixel((0, 0))
        assert actual == c, f"initial color failed for mode {mode}, expected {c} got {actual}"
```

File: Tests/test_image_access.py [hunk 3]
```diff
@@ -166,13 +169,15 @@ def check(self, mode, c=None):
...
+
         # check initial color negative index
-        assert (
-            im.getpixel((-1, -1)) == c
-        ), f"initial color failed with negative index for mode {mode}, color {c} "
+        d = im.getpixel((-1, -1))
+        assert d == c, (
+            f"initial color failed with negative index for mode {mode}, "
+            f"expected {c} got {d}"
+        )
```
Comment (hugovk) [inline Tests/test_image_access.py:180]:
```suggestion
        actual = im.getpixel((-1, -1))
        assert actual == c, (
            f"initial color failed with negative index for mode {mode}, "
            f"expected {c} got {actual}"
        )
```

**Review by hugovk — APPROVED** [review]
Thank you!

**Comment by radarhere** [comment]
If I break the test in main, I get
```
E       AssertionError: put/getpixel roundtrip failed for mode I;16B, color 32766
E       assert 32767 == 32766
E        +  where 32767 = <bound method Image.getpixel of <PIL.Image.Image image mode=I;16B size=1x1 at 0x104945F70>>((0, 0))
E        +    where <bound method Image.getpixel of <PIL.Image.Image image mode=I;16B size=1x1 at 0x104945F70>> = <PIL.Image.Image image mode=I;16B size=1x1 at 0x104945F70>.getpixel
```

If I break the test with this branch, I get
```
E       AssertionError: put/getpixel roundtrip failed for mode I;16, expected 32766 got 32767
E       assert 32767 == 32766
```

I can see that the argument that the 'where' lines being removed makes it tidier, but I'm not convinced that the values need to be part of the error message, as they're already listed in the next line?

It is listed in the summary now as
```
FAILED Tests/test_image_access.py::TestCffiGetPixel::test_signedness[32769-I;16B] - AssertionError: put/getpixel roundtrip failed for mode I;16B, expected 32768 got 32769
```
where the actual value wouldn't be listed otherwise. I'm not opposed to that, I just wonder if this is trying to fight against pytest's decision to have the summary not include more details.

**Comment by Yay295** [comment]
The only change is from "color 32766" to "expected 32766 got 32767". I don't think that's excessive for an error message. Also, if a message isn't provided to the `assert`, the summary message will be just "assert 32767 == 32766". So the current summary actually gives less information than the default.

**Pull request merged**